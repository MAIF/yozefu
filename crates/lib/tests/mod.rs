use insta::{assert_debug_snapshot, glob};
use rdkafka::message::OwnedMessage;
use serde::Deserialize;
use std::fs;
use tokio::runtime::Runtime;
use yozefu_lib::{ExportedKafkaRecord, KafkaRecord, parse_search_query};

#[test]
fn test_inputs() {
    unsafe {
        use std::env;
        // Set the timezone to Paris to have a fixed timezone for the tests
        env::set_var("TZ", "Europe/Paris");
    }
    glob!("inputs/search-queries/*.sql", |path| {
        let input = fs::read_to_string(path).unwrap();
        let input = input.trim();
        insta::with_settings!({
            description => input.replace("\n", " "),
            filters => vec![
            ("[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}\\.[0-9]{6,}\\+[0-9]{2}:[0-9]{2}", "[datetime]"),
        ]}, {
            assert_debug_snapshot!(parse_search_query(input));
        });
    });
}

#[test]
fn test_exported_record() {
    glob!("inputs/parsed-records/record*.json", |path| {
        let input = fs::read_to_string(path).unwrap();
        let record: KafkaRecord = serde_json::from_str(&input).unwrap();
        assert_debug_snapshot!(ExportedKafkaRecord::from(&record));
    });
}

#[test]
fn test_parse_records() {
    let rt = Runtime::new().unwrap();
    glob!("inputs/raw-records/record*.json", |path| {
        let input = fs::read_to_string(path).unwrap();
        let key_value: KeyValue = serde_json::from_str(&input).unwrap();
        let owned_message = OwnedMessage::new(
            key_value.value,
            key_value.key,
            "my-topic".to_string(),
            rdkafka::Timestamp::CreateTime(0),
            0,
            0,
            None,
        );
        rt.block_on(async {
            assert_debug_snapshot!(KafkaRecord::parse(owned_message, &mut None).await);
        });
    });
}

#[derive(Clone, Debug, Deserialize, Hash, PartialEq, Eq, Default)]
#[serde(rename_all = "lowercase")]
struct KeyValue {
    pub key: Option<Vec<u8>>,
    pub value: Option<Vec<u8>>,
}
