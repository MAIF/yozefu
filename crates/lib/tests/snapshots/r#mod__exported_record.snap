---
source: crates/lib/tests/mod.rs
expression: "ExportedKafkaRecord::from(&record)"
input_file: crates/lib/tests/inputs/parsed-records/record-1.json
---
ExportedKafkaRecord {
    record: KafkaRecord {
        topic: "public-french-addresses",
        timestamp: Some(
            1732481153241,
        ),
        partition: 0,
        offset: 4,
        headers: {},
        key_schema: Some(
            Schema {
                id: SchemaId(
                    1,
                ),
                schema_type: None,
            },
        ),
        value_schema: Some(
            Schema {
                id: SchemaId(
                    2,
                ),
                schema_type: None,
            },
        ),
        size: 381,
        key: Json(
            String("381"),
        ),
        key_as_string: "",
        value: Json(
            Object {
                "type": String("Feature"),
                "geometry": Object {
                    "type": String("Point"),
                    "coordinates": Array [
                        Number(4.0215),
                        Number(49.211613),
                    ],
                },
                "properties": Object {
                    "label": String("Rue Franz Kafka 51100 Reims"),
                    "score": Number(0.6992354545454544),
                    "id": String("51454_3782"),
                    "name": String("Rue Franz Kafka"),
                    "postcode": String("51100"),
                    "citycode": String("51454"),
                    "x": Number(774438.44),
                    "y": Number(6901804.03),
                    "city": String("Reims"),
                    "context": String("51, Marne, Grand Est"),
                    "type": String("street"),
                    "importance": Number(0.69159),
                    "street": String("Rue Franz Kafka"),
                },
            },
        ),
        value_as_string: "",
    },
    date_time: Some(
        2024-11-24T21:45:53.241+01:00,
    ),
    absolute_delta_in_ms: 0,
    relative_delta_in_ms: 0,
    search_query: "",
}
